Added filters responsiveness in mobile view

Ciggzy1312 opened the pull request:
Signed-off-by: Deepayan <deepayanmukherjee1312@gmail.com>

**Description**

This PR fixes #2439

**Notes for Reviewers**


**[Signed commits](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.
 

<!--
Thank you for contributing to Layer5 projects! 

Contributing Conventions:

1. Include descriptive PR titles with [<component-name>] prepended.
2. Build and test your changes before submitting a PR. 
3. Sign your commits

By following the community's contribution conventions upfront, the review process will 
be accelerated and your PR merged more quickly.
-->

l5io commented:
🚀 Preview for commit 7046fb1ab943e347d85fd62c1600b3d34afd0845 at: https://61eb0de18995c15b9f6cabf4--layer5ng.netlify.app

debo19 reviewed (changes requested):
Good work @Ciggzy1312. Couple of suggestions here:
- The `clear filters` option is not visible in desktop view now. 
- The button is not very distinguishable from the rest of the text at the moment. Try using our custom green button ( eg. the Subscribe one in the footer) to enclose this.
- Instead of "Close Filters" it can say "Apply filters".

- So if you're removing the "Filters" heading, the `clear filters` option appears right beside this "Close filters" button, which is not favourable. We can do one of two things. Either keep the "Filters" heading as it is even in mobile view without any style changes or add another button beside "Apply filters" to "Clear Filters" instead of what we have currently.

Ciggzy1312 commented:
https://user-images.githubusercontent.com/65884232/151586388-7ac363ef-2ec6-4e84-b22c-6515b39a37a1.mp4

@debo19 does it look okay?

- Fixed the `clear filters` option in desktop view
- Converted the filters tag into a button rather than keeping open or close filters
- Keeping the `clear filters` as it was because to me it looks good with the new `filters` button and making it a button might make everything clumsy(Need opinions on this)

If the overview looks good enough I might commit the changes for a review

debo19 commented:
@Ciggzy1312 The button looks much better. You can push your changes so we can review further. Also, merge conflicts to address 😅

l5io commented:
🚀 Preview for commit 76a337733ddc554874fc717b0afbecc3cc3007c5 at: https://61f6c901290c89545b304d8e--layer5ng.netlify.app

debo19 reviewed (commented):
Good work @Ciggzy1312. Couple of things to address:
1.  The filter options should be open by default in desktop view but closed in mobile view when opened for the first time.
2.  The hover effect on clear filters button is removed, is that intentional?
3.   This space needs to be decreased slightly.
<img width="347" alt="Screenshot 2022-02-10 at 7 25 59 PM" src="https://user-images.githubusercontent.com/85789734/153422712-0d2ecf15-e0aa-4bc2-80c1-0a2b81817fc9.png">

4. The "x Filters" button can be misguiding as it looks like a remove filters option. Saying "Apply Filters" and dropping the cross may be appropriate.
5.  Can we have something like this? The two buttons can be side by side or vertical, see which fits better. What do you think? <img width="239" alt="Screenshot 2022-02-10 at 7 33 13 PM" src="https://user-images.githubusercontent.com/85789734/153423438-0bbcd710-d945-460a-a7f5-829851930353.png">

l5io commented:
🚀 Preview for commit ce36ccc64fdd8f47212a2a94379b643af61b5385 at: https://62124662335c6a2943518d2b--layer5ng.netlify.app

leecalcote commented:
Functionally, this is working great.

debo19 commented on src/sections/Resources/Resources-grid/filters.js:
```diff
@@ -74,132 +77,160 @@ const Navigation = (props) => {
+            <span className= {`${props.resources.length === 0 ? "clear-disabled" : "clear-enabled"} desk-view`} onClick={props.clear}>
+              Clear Filters 
+            </span>
+            {expandFilter ? (<span className= {`${props.resources.length === 0 ? "clear-disabled" : "clear-enabled mob-view"}`} onClick={props.clear}>
```
Since the only difference is the `className`, we shouldn't need to have `Clear Filters` defined twice.

debo19 replied:
```suggestion
            {expandFilter ? (<span className= {
            `${props.resources.length === 0 ? "clear-disabled" : 
            expandFilter ? "clear-enabled mob-view" : "clear-enabled desktop-view"}
            `} 
            onClick={props.clear}>
```
Something like this?

Ciggzy1312 replied:
Putting them together won't work unfortunately
The "desk-view" is put separately and doesn't depend on the resources length where as the "mob-view" does
Tried many times but this is the only way I could make it work as of now

debo19 reviewed (commented):
Looking good @Ciggzy1312. Just a small change.

debo19 reviewed (approved):
LGTM 🎉 Thanks for seeing this through @Ciggzy1312
